godot-core/src/builtin/array.rs

```diff
@@ -619,13 +676,35 @@ impl<T: GodotType + ToGodot> Array<T> {
     /// On large arrays, this method is much slower than `pop_back` as it will move all the array's
     /// elements after the removed element. The larger the array, the slower `remove` will be.
     pub fn erase(&mut self, value: &T) {
-        self.as_inner().erase(value.to_variant());
+        // SAFETY: We don't write anything to the array.
+        unsafe { self.as_inner() }.erase(value.to_variant());
     }
 
     /// Assigns the given value to all elements in the array. This can be used together with
     /// `resize` to create an array with a given size and initialized elements.
     pub fn fill(&mut self, value: &T) {
-        self.as_inner().fill(value.to_variant());
+        // SAFETY: The array has type `T` and we're writing values of type `T` to it.
+        unsafe { self.as_inner() }.fill(value.to_variant());
+    }
+
+    /// Resizes the array to contain a different number of elements.
+    ///
+    /// If the new size is smaller than the current size, then it removes elements from the end. If the new size is bigger than the current one
+    /// then the new elements are set to `value`.
+    ///
+    /// If you know that the new size is smaller, then consider using [`shrink`](Array::shrink) instead.
+    #[doc(alias = "resize")]
+    pub fn resize_with(&mut self, new_size: usize, value: &T) {
+        let original_size = self.len();
+        // SAFETY: While we do insert `Variant::nil()` if the new size is larger, we then fill it with `value` ensuring that all values in the
+        // array are of type `T` still.
+        unsafe { self.as_inner() }.resize(to_i64(new_size));
+
+        if new_size > original_size {
+            for i in original_size..new_size {
+                self.set(i, value.to_godot());
+            }
+        }
     }
```

Comment:
1. I think it would be good to declare this method next to `shrink`, so they are co-located in the docs.

2. The `_with` suffix suggests that the new element takes a closure, see [`Vec::resize_with`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.resize_with). I think it's fair to call it `resize` (like `Vec::resize`), even if the signature differs from Godot.

3. The `if new_size > original_size` is not necessary, `original_size..new_size` will be an empty iterator if it is not larger. A comment might help clarify that though.